build.psm1

```diff
@@ -286,122 +287,191 @@ function Start-PSBuild {
 "@
-                log "  Executing Build Command: $command"
-                Start-NativeExecution { Invoke-Expression -Command:$command }
+                    log "  Executing Build Command: $command"
+                    Start-NativeExecution { Invoke-Expression -Command:$command }
 
-                $clrTarget = "FullClr"
-                if ($OneCoreValue -eq "ON")
-                {
-                    $clrTarget = "CoreClr"
+                    $clrTarget = "FullClr"
+                    if ($OneCoreValue -eq "ON")
+                    {
+                        $clrTarget = "CoreClr"
+                    }
+
+                    # Copy the binaries from the local build directory to the packaging directory
+                    $dstPath = ($script:Options).Top
+                    $FilesToCopy | % {
+                        $srcPath = Join-Path (Join-Path (Join-Path (Get-Location) "bin") $msbuildConfiguration) "$clrTarget/$_"
+                        log "  Copying $srcPath to $dstPath"
+                        Copy-Item $srcPath $dstPath
+                    }
                 }
 
-                # Copy the binaries from the local build directory to the packaging directory
-                $dstPath = ($script:Options).Top
-                $FilesToCopy | % {
-                    $srcPath = Join-Path (Join-Path (Join-Path (Get-Location) "bin") $msbuildConfiguration) "$clrTarget/$_"
-                    log "  Copying $srcPath to $dstPath"
-                    Copy-Item $srcPath $dstPath
+                if ($FullCLR) {
+                    $fullBinaries = @(
+                        'powershell.exe',
+                        'powershell.pdb',
+                        'pwrshplugin.dll',
+                        'pwrshplugin.pdb'
+                    )
+                    Build-NativeWindowsBinaries "OFF" $fullBinaries
+                }
+                else
+                {
+                    $coreClrBinaries = @(
+                        'pwrshplugin.dll',
+                        'pwrshplugin.pdb'
+                    )
+                    Build-NativeWindowsBinaries "ON" $coreClrBinaries
                 }
+            } finally {
+                Pop-Location
             }
-
-            if ($FullCLR) {
-                $fullBinaries = @(
-                    'powershell.exe',
-                    'powershell.pdb',
-                    'pwrshplugin.dll',
-                    'pwrshplugin.pdb'
+        }
+    }
+    else
+    {
+        # use native binaries from $PSHOME, assume it is latest
+        $nativeBinaries = @()
+        if ($IsWindows)
+        {
+            if ($PSVersionTable.PSEdition -ne "Core")
+            {
+                # if running under Windows PowerShell, we'll need to get the binaries from GitHub release
```

Comment:
At this point, should we maybe just build the native binaries rather than download them? I get for the rest of the scenarios this PR makes us use existing native binaries on the machine, but here, in Windows PowerShell, why download a PowerShell package over building the binaries?

I could see a reason: installing the tools to build the native binaries on Windows can be an annoying task, but perhaps we should document the reason we avoid this.